gapic/src/main/com/google/gapid/util/Logging.java

```diff
@@ -15,54 +15,64 @@
  */
 package com.google.gapid.util;
 
+import com.google.gapid.proto.log.Log;
+import com.google.gapid.rpclib.schema.Method;
 import com.google.gapid.util.Flags.Flag;
 import com.google.protobuf.MessageOrBuilder;
+import com.google.protobuf.Timestamp;
 
 import java.io.File;
 import java.io.IOException;
 import java.io.PrintWriter;
 import java.io.StringWriter;
 import java.text.SimpleDateFormat;
 import java.util.Date;
-import java.util.logging.ErrorManager;
-import java.util.logging.FileHandler;
-import java.util.logging.Formatter;
-import java.util.logging.Handler;
-import java.util.logging.Level;
-import java.util.logging.LogManager;
-import java.util.logging.LogRecord;
-import java.util.logging.Logger;
-import java.util.logging.StreamHandler;
+import java.util.LinkedList;
+import java.util.logging.*;
 
 /**
  * Logging setup and utilities.
  */
 public class Logging {
   /**
-   * Possible log level flag values.
+   * Possible logMessage level flag values.
    */
-  public static enum LogLevel {
-    OFF(Level.OFF, "Emergency", "F"), ERROR(Level.SEVERE, "Error", "E"),
-    WARNING(Level.WARNING, "Warning", "W"), INFO(Level.INFO, "Info", "I"),
-    DEBUG(Level.FINE, "Debug", "D"), ALL(Level.ALL, "Debug", "V");
+  public enum LogLevel {
+    OFF(Level.OFF, Log.Severity.Fatal, "Fatal", "F"),
+    ERROR(Level.SEVERE, Log.Severity.Error, "Error", "E"),
+    WARNING(Level.WARNING, Log.Severity.Warning, "Warning", "W"),
+    INFO(Level.INFO, Log.Severity.Info, "Info", "I"),
+    DEBUG(Level.FINE, Log.Severity.Debug, "Debug", "D"),
+    ALL(Level.ALL, Log.Severity.Debug, "Debug", "V");
 
     public final Level level;
+    public final Log.Severity severity;
     public final String gapisLevel;
     public final String gapirLevel;
 
-    private LogLevel(Level level, String gapisLevel, String gapirLevel) {
+    LogLevel(Level level, Log.Severity severity, String gapisLevel, String gapirLevel) {
       this.level = level;
+      this.severity = severity;
       this.gapisLevel = gapisLevel;
       this.gapirLevel = gapirLevel;
     }
+
+    static LogLevel fromLevel(Level level) {
+      for (LogLevel logLevel : LogLevel.values()) {
+        if (logLevel.level == level) {
```

Comment:
This should be range based as it was before.
java.util.logging.Level is not an enum, so you also should not use '==', but .equals.